src/libutil/linux/cgroup.cc

```diff
@@ -101,6 +102,11 @@ static CgroupStats destroyCgroup(const std::filesystem::path & cgroup, bool retu
             // FIXME: pid wraparound
             if (kill(pid, SIGKILL) == -1 && errno != ESRCH)
                 throw SysError("killing member %d of cgroup '%s'", pid, cgroup);
+
+            while (waitpid(pid, nullptr, 0) == -1) {
```

Comment:
Not sure if it matters, but I think this only works for processes that are direct children of the current process? For others it will return `ECHILD` according to the manpage.

Reply:
It also seems like it doesn't work correctly because there some other code that uses waitpid and crashes if there is no wait result.